Src/FluentAssertions/Equivalency/DataColumnEquivalencyStep.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+
+using FluentAssertions.Data;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Equivalency
+{
+    public class DataColumnEquivalencyStep : IEquivalencyStep
+    {
+        /// <summary>
+        /// Gets a value indicating whether this step can handle the current subject and/or expectation.
+        /// </summary>
+        public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAssertionOptions config)
+        {
+            return typeof(DataColumn).IsAssignableFrom(config.GetExpectationType(context));
+        }
+
+        /// <summary>
+        /// Applies a step as part of the task to compare two objects for structural equality.
+        /// </summary>
+        /// <value>
+        /// Should return <c>true</c> if the subject matches the expectation or if no additional assertions
+        /// have to be executed. Should return <c>false</c> otherwise.
+        /// </value>
+        /// <remarks>
+        /// May throw when preconditions are not met or if it detects mismatching data.
+        /// </remarks>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0019:Use pattern matching", Justification = "The code is easier to read without it.")]
+        public bool Handle(IEquivalencyValidationContext context, IEquivalencyValidator parent, IEquivalencyAssertionOptions config)
```

Comment:
🙃 This is quite a long method. I would myself try to refactor it a bit.

Reply:
Wait, the DataColumn one is long? 😮 Oh dear 😅

Reply:
I have split apart long methods in all of the equivalency step implementations.